client/src/components/uis/ChannelListItem.tsx

```diff
@@ -125,35 +150,73 @@ interface Props {
   fontColor?: string;
 }
 
+class FilteredMemberships {
+  constructor(
+    private memberships: Maybe<Membership[]> | undefined,
+    private user: AuthProviderMeQueryResponse['me'] | null,
+    private channelType: string,
+  ) {
+    this.memberships = memberships;
+    this.channelType = channelType;
+    this.user = user;
+  }
+
+  private get filtered(): (Maybe<Membership> | undefined)[] | undefined {
+    return this.memberships?.filter((member) => {
+      if (this.channelType !== 'self')
+        return member?.user?.id !== this.user?.id;
+
+      return true;
+    });
+  }
+
+  get users(): (Maybe<User> | undefined)[] | undefined {
+    return this.filtered?.map((membership) => membership?.user);
+  }
+
+  get userNames(): string[] | undefined {
+    return this.users?.map((v) => v?.nickname || v?.name || '');
+  }
+  calculatedPhotoUrls(): (Maybe<string> | undefined)[] | undefined {
+    return this.filtered?.map(
+      (membership) => membership?.user?.thumbURL || membership?.user?.photoURL,
+    );
+  }
+
+  calculateOnlineStatus(): Maybe<Boolean> | undefined {
+    return this.filtered?.[0]?.user?.isOnline;
+  }
+
+  calculateDisplayName(): Maybe<string> | undefined {
+    return this.users?.length === 1
+      ? this.users?.[0]?.nickname ||
+          this.users?.[0]?.name ||
+          getString('NO_NAME')
+      : this.userNames?.join(', ');
+  }
+}
```

Comment:
I think functions are simpler than a class. Something like this:
```ts
const calculateUsers = (memberships: Membership[], channelType: string, authUserId): (Maybe<User>)[] => 
  memberships.filter((member) => channelType !== 'self' && member?.user?.id !== authUserId);

const calculatePhotoUrls = (memberships: Membership[]): (Maybe<string>)[] =>
  memberships.map((member) => member?.user?.thumbURL || member?.user?.photoURL);

const calculateOnlineStatus = (users: (Maybe<User>)[]): boolean =>
  users[0]?.isOnline ?? false;
```

Reply:
Indeed, photoUrls also needs 'filtered by channel type' `memberships` passed as an argument, So there needs to be `filteredMemberships` calculated first and `filteredMemberships' is only used for this functions/class method then I just think that using class is better for reusability or further possible use cases.